examples/multi-gpu-movielens/01-03b-MultiGPU-Download-Convert-ETL-with-NVTabular-Training-with-PyTorch.ipynb

```diff
@@ -0,0 +1,683 @@
+    "- Use multi-GPU or multi node for ETL with NVTabular\n",
+    "- Use NVTabular dataloader to accelerate PyTorch pipelines\n",
+    "- Scale PyTorch training with Horovod\n",
+    "\n",
+    "### Dataset\n",
+    "\n",
+    "In this notebook, we use the [MovieLens25M](https://grouplens.org/datasets/movielens/25m/) dataset. It is popular for recommender systems and is used in academic publications. The dataset contains 25M movie ratings for 62,000 movies given by 162,000 users. Many projects use only the user/item/rating information of MovieLens, but the original dataset provides metadata for the movies, as well.\n",
+    "\n",
+    "Note: We are using the MovieLens 25M dataset in this example for simplicity, although the dataset is not large enough to require multi-GPU training. However, the functionality demonstrated in this notebook can be easily extended to scale recommender pipelines for larger datasets in the same way.\n",
+    "\n",
+    "### Tools\n",
+    "\n",
+    "- [Horovod](https://horovod.readthedocs.io/en/stable/) is a distributed deep learning framework that provides tools for multi-GPU optimization.\n",
+    "- The [NVIDIA Collective Communication Library (NCCL)](https://developer.nvidia.com/nccl) provides the underlying GPU-based implementations of the [allgather](https://docs.nvidia.com/deeplearning/nccl/user-guide/docs/usage/operations.html#allgather) and [allreduce](https://docs.nvidia.com/deeplearning/nccl/user-guide/docs/usage/operations.html#allreduce) cross-GPU communication operations."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "1a4d934b",
+   "metadata": {},
+   "source": [
+    "## Download and Convert\n",
+    "\n",
+    "First, we will download and convert the dataset to Parquet. This section is based on [01-Download-Convert.ipynb](../getting-started-movielens/01-Download-Convert.ipynb)."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "dff780d1",
+   "metadata": {},
+   "source": [
+    "#### Download"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "id": "e21689a8",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# External dependencies\n",
+    "import cudf                 # cuDF is an implementation of Pandas-like Dataframe on GPU\n",
+    "import time\n",
+    "import gc\n",
+    "import pathlib\n",
+    "import zipfile\n",
+    "\n",
+    "from urllib import request\n",
+    "from sklearn.model_selection import train_test_split\n",
+    "\n",
+    "BASE_DIR = './data'\n",
+    "\n",
+    "zip_path = pathlib.Path(BASE_DIR, 'ml-25m.zip')\n",
+    "\n",
+    "if not zip_path.exists():\n",
+    "    zip_path.parent.mkdir(parents=True, exist_ok=True)\n",
+    "    \n",
+    "    request.urlretrieve(\"http://files.grouplens.org/datasets/movielens/ml-25m.zip\", str(zip_path))\n",
+    "    \n",
+    "    with zipfile.ZipFile(zip_path,\"r\") as zip_ref:\n",
+    "        zip_ref.extractall(BASE_DIR)"
```

Comment:
I added a utility function in https://github.com/NVIDIA/NVTabular/pull/709 to download and unzip files. It's pretty much the same as here, except includes progress bars of download progress.

Reply:
I rewrote this to use `pathlib` instead of `os` because `os` calls were failing for me (which are also used in the `download_file` util function.) I'm happy to standardize this code and stop having to write it all over the place, but I'm not sure `download_file` addresses my issue. I'll give it a try.

Reply:
I do like progress bars! :smile:

Reply:
what was the error with the os.path calls? I'm surprised that it doesn't work (like I think pathlib has better ergonomics / and is a more modern way of manipulating paths and since we only work on python 3.6+ we should use - but I didn't realize it was doing any thing fundamentally different than os.path). 

Also, We don't need to change for this release - we can merge as is (and maybe standardize and upgrade download_file etc to also use pathlib etc).